### ISSUE TITLE
Unable to build for freebsd arm v7

### ISSUE THREAD

**Issue by efficks** [shirou/gopsutil#627]
**Describe the bug**
When building for freebsd arm7 cpu, I got compilation errors 

```
# github.com/influxdata/telegraf/vendor/github.com/shirou/gopsutil/disk
vendor/github.com/shirou/gopsutil/disk/disk_freebsd.go:154:9: undefined: Bintime
# github.com/influxdata/telegraf/vendor/github.com/shirou/gopsutil/cpu
vendor/github.com/shirou/gopsutil/cpu/cpu_freebsd.go:25:16: undefined: cpuTimes
vendor/github.com/shirou/gopsutil/cpu/cpu_freebsd.go:42:31: undefined: cpuTimes
vendor/github.com/shirou/gopsutil/cpu/cpu_freebsd.go:66:38: undefined: cpuTimes
vendor/github.com/shirou/gopsutil/cpu/cpu_freebsd.go:72:15: undefined: cpuTimes
vendor/github.com/shirou/gopsutil/cpu/cpu_freebsd.go:87:13: undefined: cpuTimes
```

**To Reproduce**
`env GOOS=freebsd GOARCH=arm GOARM=7 go`

**Expected behavior**
Compilation success

**Environment (please complete the following information):**
I'm cross compiling on a linux 64 bit for freebsd arm v7.

**Additional context**
go env
```
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/fxchoiniere/.cache/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/fxchoiniere/go"
GORACE=""
GOROOT="/usr/lib/go-1.10"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go-1.10/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build088163185=/tmp/go-build -gno-record-gcc-switches"
```

**Comment by Lomanic** [comment]
Copying the i386 definition to arm successfully compiles (`cp cpu/cpu_freebsd_386.go cpu/cpu_freebsd_arm.go`), tested with `GOOS=freebsd GOARCH=arm go build github.com/shirou/gopsutil/cpu`. Can you test that the telegraph binary you will get doesn't crash at runtime on a FreeBSD ARM host? It should still work as both architectures are 32 bit but just to be sure.

**Comment by efficks** [comment]
@Lomanic It works like a charm. I also copied the disk_freebsk_386 to arm.
Thank you

**Comment by Lomanic** [comment]
https://github.com/golang/go/wiki/GoArm#supported-operating-systems

> ARM on FreeBSD, OpenBSD, and NetBSD: ARMv6K or above is required.

I still have the plan, for months now, to install FreeBSD on a Rapsberry Pi Model B 512 MB (armv6) to properly fix #543 (and this issue now too). I just did not manage to get around it in my limited free time. Keep in mind that an armv7 host is able to run an armv6 binary without problem (only lacking some optimizations). For arm64, there is [this path on Scaleway](https://community.scaleway.com/t/freebsd-on-arm64/6678) or buying a Pine64 board.

**Comment by danielnelson** [comment]
I'm guessing you are hoping to generate the cgo file there in case a simple copy is not completely correct?  If so, I think @W0CHP mentioned an arm system with a working buildchain?

**Comment by shirou** [comment]
I have created a PR to add AMR support files. But generate by using cgo on real system is better than just a copy. If you can run cgo on your system please make a comment on #632. Thank you!

**Comment by shirou** [comment]
perhaps this is solved by #632. feel free to re-open this issue if something wrong.
